include/hal/library/cryptlib/cryptlib_cert.h

```diff
@@ -412,7 +412,7 @@ extern bool libspdm_sm2_get_public_key_from_x509(const uint8_t *cert, size_t cer
 extern bool libspdm_gen_x509_csr(size_t hash_nid, size_t asym_nid,
                                  uint8_t *requester_info, size_t requester_info_length,
                                  void *context, char *subject_name,
-                                 size_t *csr_len, uint8_t **csr_pointer);
+                                 size_t *csr_len, uint8_t *csr_pointer);
```

Comment:
I just noticed that `libspdm_gen_x509_csr` is not used by any of the core libraries. Please move this declaration to https://github.com/DMTF/libspdm/blob/main/os_stub/spdm_crypt_ext_lib/cryptlib_ext.h